src/dictionary/dictionary_types.h

```diff
@@ -421,6 +421,13 @@ typedef struct hash_bucket {
 	ion_byte_t	data[];			/**< the data in the bucket */
 } ion_hash_bucket_t;
 
+/**
+@brief		Function signature for all init functions.
+*/
+typedef void (*init_t)(
```

Comment:
Lets name this `ion_init_t` - or maybe `ion_handler_initializer_t`? I leave the choice up to you